src/Stratis.Features.FederatedPeg/Wallet/MultiSigTransactions.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg.Wallet
+{
+    public class MultiSigTransactions : ICollection<TransactionData>, ITransactionDataObserver
```

Comment:
Nice.

It's important that this class is always returning accurate data, even as transactions get added, confirmed, and spent, as this wasn't so complex before. 

Might be good to have some tests purely on this class for adding and updating transactions and always getting back correct data.

Reply:
Agree.